### PR TITLE
fix(feedback): add schema fixes back in

### PR DESCRIPTION
**Pull request by JoshFerge**
add in schema fixes removed in https://github.com/getsentry/sentry/pull/59357

adds regression test.

### CODE REVIEW

**Review by c298lee — APPROVED** [review]

File: src/sentry/feedback/usecases/create_feedback.py [hunk 1]
```diff
@@ -73,6 +73,17 @@ def fix_for_issue_platform(event_data):
     ret_event["request"] = event_data.get("request", {})
 
     ret_event["user"] = event_data.get("user", {})
+
+    if event_data.get("dist") is not None:
```
Comment (cmanallen) [inline src/sentry/feedback/usecases/create_feedback.py:77]: `event_data.pop("dist", None)`

File: src/sentry/feedback/usecases/create_feedback.py [hunk 2]
```diff
@@ -73,6 +73,17 @@ def fix_for_issue_platform(event_data):
...
+        del event_data["dist"]
+    if event_data.get("user", {}).get("name") is not None:
```
Comment (cmanallen) [inline src/sentry/feedback/usecases/create_feedback.py:79]: `event_data.get("user", {}).pop("name", None)`

File: src/sentry/feedback/usecases/create_feedback.py [hunk 3]
```diff
@@ -73,6 +73,17 @@ def fix_for_issue_platform(event_data):
...
+        del event_data["user"]["name"]
+    if event_data.get("user", {}).get("isStaff") is not None:
```
Comment (cmanallen) [inline src/sentry/feedback/usecases/create_feedback.py:81]: `event_data.get("user", {}).pop("isStaff", None)`

**Review by billyvg — APPROVED** [review]

File: src/sentry/feedback/usecases/create_feedback.py (hunk 1 above)
Comment (cmanallen) [reply src/sentry/feedback/usecases/create_feedback.py:77]: If you want the key to continue existing if its value is `None` then this option (and the following options) won't work.

**Review by cmanallen — APPROVED** [review]

**Pull request merged**